src/jit/codegenarmarch.cpp

```diff
@@ -666,11 +672,21 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 var_types type = nextArgNode->TypeGet();
                 emitAttr  attr = emitTypeSize(type);
 
+#ifdef _TARGET_ARM64_
+                // Emit store instructions to store the registers produced by the GT_FIELD_LIST into the outgoing
+                // argument area
+                emit->emitIns_S_R(ins_Store(type), attr, reg, varNumOut, argOffsetOut + fieldListPtr->gtFieldOffset);
+
+                // We can't write beyound the outgoing area area
+                assert((argOffsetOut + fieldListPtr->gtFieldOffset + EA_SIZE_IN_BYTES(attr)) <= argOffsetMax);
+#else
+                // TODO-ARM-Bug?  The following code will pack copied structs
```

Comment:
@BruceForstall
 We need to test this code path on ARM32

Reply:
@briansull - this is simply the existing code path for ARM32 (it doesn't use the `gtFieldOffset` as set in `fgMorphLclArgToFieldlist()`) so I'm not sure why it needs to be specially tested.
However, we do need to determine if the existing code was actually correct - is that what you meant?

Reply:
Yes,  I meant - determine if the existing code was actually correct.
If this is only enabled when FEATURE_MULTIREG_STRUCT_PROMOTE then it is ifdef-ed out.
As Steve says it appears to be incorrect, as it will perform mis-aligned stores.